src/activity_actor.cpp

```diff
@@ -387,11 +579,20 @@ void open_gate_activity_actor::finish( player_activity &act, Character & )
     act.set_to_null();
 }
 
+/** @pre @p other is an open_gate_activity_actor */
+bool open_gate_activity_actor::can_resume_with( const activity_actor &other,
+        const Character & ) const
+{
+    const open_gate_activity_actor &og_actor = dynamic_cast<const open_gate_activity_actor &>( other );
+
+    return *this == og_actor;
+}
+
 void open_gate_activity_actor::serialize( JsonOut &jsout ) const
 {
     jsout.start_object();
 
-    jsout.member( "moves", moves );
+    jsout.member( "moves_total", moves_total );
```

Comment:
Changing the JSON key would break old saves. You should either keep the key as "moves", or read "moves" if "moves_total" is not found.

Reply:
Thanks, I did not consider that.  I think keeping the key as "moves" would work fine. The reason I wanted to change the variable to moves_total was because once I started to consider resuming, I felt like it should be clear that "moves" is not keeping track of moves left.  I think it might also be good for me to change the key to "moves" for both dig actors as well to be consistent (but that should not break any saves as this is new code).